sdk/identity/azure-identity/test/ut/token_credential_impl_test.cpp

```diff
@@ -663,3 +666,313 @@ TEST(TokenCredentialImpl, ExpirationFormats)
+        "absolute expiration property ('TokenExpiresAtTime') is NOT present, "
+        "and there are no other properties.");
+
+    Logger::SetListener(nullptr);
+  }
+
+  {
+    LogMsgVec log;
+    Logger::SetListener([&](auto lvl, auto msg) { log.push_back(std::make_pair(lvl, msg)); });
+
+    try
+    {
+      static_cast<void>(TokenCredentialImpl::ParseToken(
+          "{\"TokenForAccessing\":\"ACCESSTOKEN\"}",
+          "TokenForAccessing",
+          "TokenExpiresInSeconds",
+          "TokenExpiresAtTime"));
+    }
+    catch (std::exception const& e)
+    {
+      EXPECT_EQ(
+          e.what(),
+          std::string("Token JSON object: can't find or parse 'TokenExpiresAtTime' property."
+                      "\nSee Azure::Core::Diagnostics::Logger for details"
+                      " (https://aka.ms/azsdk/cpp/identity/troubleshooting)."));
+    }
+
+    EXPECT_EQ(log.size(), 1U);
+    EXPECT_EQ(log.at(0).first, Logger::Level::Verbose);
+    EXPECT_EQ(
+        log.at(0).second,
+        "Identity: TokenCredentialImpl::ParseToken(): "
+        "Token JSON: Access token property ('TokenForAccessing') is string (length=11), "
+        "relative expiration property ('TokenExpiresInSeconds') is NOT present, "
+        "absolute expiration property ('TokenExpiresAtTime') is NOT present, "
+        "and there are no other properties.");
+
+    Logger::SetListener(nullptr);
+  }
+
+  {
+    LogMsgVec log;
+    Logger::SetListener([&](auto lvl, auto msg) { log.push_back(std::make_pair(lvl, msg)); });
+
+    try
+    {
+      static_cast<void>(TokenCredentialImpl::ParseToken(
+          "{\"TokenForAccessing\":\"ACCESSTOKEN\","
+          "\"TokenExpiresInSeconds\":\"one\""
+          "}",
+          "TokenForAccessing",
+          "TokenExpiresInSeconds",
+          {}));
+    }
+    catch (std::exception const& e)
+    {
+      EXPECT_EQ(
+          e.what(),
+          std::string("Token JSON object: can't find or parse 'TokenExpiresInSeconds' property."
+                      "\nSee Azure::Core::Diagnostics::Logger for details"
+                      " (https://aka.ms/azsdk/cpp/identity/troubleshooting)."));
+    }
+
+    EXPECT_EQ(log.size(), 1U);
+    EXPECT_EQ(log.at(0).first, Logger::Level::Verbose);
+    EXPECT_EQ(
+        log.at(0).second,
+        "Identity: TokenCredentialImpl::ParseToken(): "
+        "Token JSON: Access token property ('TokenForAccessing') is string (length=11), "
+        "relative expiration property ('TokenExpiresInSeconds') is present (value='\"one\"'), "
+        "absolute expiration property ('') is NOT present, "
+        "and there are no other properties.");
+
+    Logger::SetListener(nullptr);
+  }
+
+  {
+    LogMsgVec log;
+    Logger::SetListener([&](auto lvl, auto msg) { log.push_back(std::make_pair(lvl, msg)); });
+
+    try
+    {
+      static_cast<void>(TokenCredentialImpl::ParseToken(
+          "{\"TokenForAccessing\":\"ACCESSTOKEN\","
```

Comment:
Would a string literal make sense here rather than all the quoted quote characters?

Reply:
I prefer all the formatting you get when you don't have a multi-line literal.